### PR TITLE
Store bench node socket in tmp dir

### PR DESCRIPTION
**Pull request by Anviking**
# Issue Number

#1458 

# Overview

- [x] Make sure the socket path is in a temporary directory

# Comments

### CODE REVIEW

**Review by paweljakubas — APPROVED** [review]

File: lib/byron/bench/Restore.hs [hunk 1]
```diff
@@ -183,12 +183,16 @@ main = do
...
+
     let genesisHash =
             B8.unpack
             . hex
             . getGenesisBlockHash
             $ blockchainParameters @'Mainnet
-    let socketPath = "cardano-node.socket"
+    let socketPath = tmpDir </> "cardano-node.socket"
     let args =
             [ "run"
```
Comment (KtorZ) [inline lib/byron/bench/Restore.hs:197]: Out of curiosity, where are the configs below coming from exactly? Are they also generated and written somewhere to disk ?

**Review by KtorZ — APPROVED** [review]

File: lib/byron/bench/Restore.hs (hunk 1 above)
Comment (Anviking) [reply lib/byron/bench/Restore.hs:197]: They are coming from https://github.com/input-output-hk/cardano-wallet/blob/65eafca208ff8c2c544e4a094f4fe3cf9b77c3b4/nix/haskell.nix#L94

**Comment by Anviking** [comment]
```

Failures:
--
  |  
  | src/Test/Integration/Framework/DSL.hs:1179:9:
  | 1) API Specifications BYRON_UPDATE_NAME_01 - Update names of wallets
  | expected: Status {statusCode = 204, statusMessage = "No Content"}
  | but got: Status {statusCode = 500, statusMessage = "Internal Server Error"}
  |  
  | from the following response: Left (HttpException ResponseTimeout)
  |  
  | To rerun use: --match "/API Specifications/BYRON_UPDATE_NAME_01 - Update names of wallets/"
  |  
```
🤔 bors r+

**Pull request merged**